File: calculator.py
"""
Simple Calculator Module
Provides basic mathematical operations
"""

class Calculator:
    """A simple calculator class with basic mathematical operations"""
    
    def __init__(self):
        """Initialize calculator with history tracking"""
        self.history = []
    
    def add(self, a, b):
        """
        Add two numbers
        Args:
            a: First number
            b: Second number
        Returns:
            Sum of a and b
        """
        result = a + b
        self.history.append(f"{a} + {b} = {result}")
        return result
    
    def subtract(self, a, b):
        """
        Subtract b from a
        Args:
            a: First number
            b: Second number
        Returns:
            Difference of a and b
        """
        result = a - b
        self.history.append(f"{a} - {b} = {result}")
        return result
    
    def multiply(self, a, b):
        """
        Multiply two numbers
        Args:
            a: First number
            b: Second number
        Returns:
            Product of a and b
        """
        result = a * b
        self.history.append(f"{a} * {b} = {result}")
        return result
    
    def divide(self, a, b):
        """
        Divide a by b
        Args:
            a: Numerator
            b: Denominator
        Returns:
            Quotient of a and b
        Raises:
            ValueError: If b is zero
        """
        if b == 0:
            raise ValueError("Cannot divide by zero")
        result = a / b
        self.history.append(f"{a} / {b} = {result}")
        return result
    
    def power(self, base, exponent):
        """
        Calculate power of a number
        Args:
            base: Base number
            exponent: Exponent value
        Returns:
            base raised to the power of exponent
        """
        result = base ** exponent
        self.history.append(f"{base} ^ {exponent} = {result}")
        return result
    
    def square_root(self, number):
        """
        Calculate square root of a number
        Args:
            number: Number to calculate square root
        Returns:
            Square root of the number
        Raises:
            ValueError: If number is negative
        """
        if number < 0:
            raise ValueError("Cannot calculate square root of negative number")
        result = number ** 0.5
        self.history.append(f"√{number} = {result}")
        return result
    
    def get_history(self):
        """
        Get calculation history
        Returns:
            List of calculation history
        """
        return self.history
    
    def clear_history(self):
        """Clear calculation history"""
        self.history = []

File: Calculator.php
<?php
/**
 * Calculator Class
 * Provides basic mathematical operations
 */
class Calculator {
    /**
     * @var array History of calculations
     */
    private $history = [];
    
    /**
     * Add two numbers
     * @param float $a First number
     * @param float $b Second number
     * @return float Sum of a and b
     */
    public function add($a, $b) {
        $result = $a + $b;
        $this->history[] = "$a + $b = $result";
        return $result;
    }
    
    /**
     * Subtract b from a
     * @param float $a First number
     * @param float $b Second number
     * @return float Difference of a and b
     */
    public function subtract($a, $b) {
        $result = $a - $b;
        $this->history[] = "$a - $b = $result";
        return $result;
    }
    
    /**
     * Multiply two numbers
     * @param float $a First number
     * @param float $b Second number
     * @return float Product of a and b
     */
    public function multiply($a, $b) {
        $result = $a * $b;
        $this->history[] = "$a * $b = $result";
        return $result;
    }
    
    /**
     * Divide a by b
     * @param float $a Numerator
     * @param float $b Denominator
     * @return float Quotient of a and b
     * @throws InvalidArgumentException If b is zero
     */
    public function divide($a, $b) {
        if ($b == 0) {
            throw new InvalidArgumentException("Cannot divide by zero");
        }
        $result = $a / $b;
        $this->history[] = "$a / $b = $result";
        return $result;
    }
    
    /**
     * Calculate power of a number
     * @param float $base Base number
     * @param float $exponent Exponent value
     * @return float Base raised to the power of exponent
     */
    public function power($base, $exponent) {
        $result = pow($base, $exponent);
        $this->history[] = "$base ^ $exponent = $result";
        return $result;
    }
    
    /**
     * Calculate square root of a number
     * @param float $number Number to calculate square root
     * @return float Square root of the number
     * @throws InvalidArgumentException If number is negative
     */
    public function squareRoot($number) {
        if ($number < 0) {
            throw new InvalidArgumentException("Cannot calculate square root of negative number");
        }
        $result = sqrt($number);
        $this->history[] = "sqrt($number) = $result";
        return $result;
    }
    
    /**
     * Get calculation history
     * @return array List of calculation history
     */
    public function getHistory() {
        return $this->history;
    }
    
    /**
     * Clear calculation history
     * @return void
     */
    public function clearHistory() {
        $this->history = [];
    }
}
